juice/src/layers/common/rnn.rs

```diff
@@ -67,29 +64,120 @@ pub enum RnnType {
     tanh,
 }
 
-impl RnnType {
+impl RnnNetworkMode_UserInput {
     fn to_text(&self) -> String {
         match self {
```

Comment:
`to_string` would be more idiomatic, and I think it'd would make more sense to impl `core::fmt::Display` instead of providing these conversions, that would give `to_string` method for free :)

Reply:
Oh _what_, I didn't realise that was free in Display, I effectively did it the other way round. Yeah I'll add that in.